zubhub_frontend/zubhub/src/components/upload_file/uploadFile.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import clsx from 'clsx';
+import ImageIcon from '@material-ui/icons/Image';
+import CustomButton from '../button/Button';
+import {
+  handleFileButtonClick,
+  handleImageFieldChange,
+} from './uploadFileScripts.js';
+import { Typography, FormControl, FormHelperText } from '@material-ui/core';
+import { getFieldAndIndex } from '../../assets/js/utils/scripts';
+
+function UploadFile(props) {
+  const {
+    id,
+    fileType,
+    uploadButtonLabel,
+    classes,
+    countFilesText,
+    multiple,
+    wraperState,
+  } = props;
+
+  const { field, index } = getFieldAndIndex(id);
+  const uploadFileRefs = {
+    fileInput: React.useRef(null),
+  };
+
+  const [state, setState] = useState({
+    media_upload: {
+      upload_dialog: false,
+      images_to_upload: [],
+      videos_to_upload: [],
+      upload_info: {},
+      upload_percent: 0,
+      uploaded_images_url: [],
+      uploaded_videos_url: [],
+    },
+  });
+
+  const handleSetUploadFileState = obj => {
+    if (obj) {
+      Promise.resolve(obj).then(obj => {
```

Comment:
Promise is not all required here, there might some other issue if needed.

Reply:
And if you hon't have additional functionality, please send the actual setState methods to children it comes in handy in multiple ways

Reply:
that was just to pass the new object to handleSetUploadFileState instead of a function each time , but as you said yesterday we will need to customize this function in some cases more than just combining the old object properties with new ones